generic_feature_generator.py

```diff
@@ -0,0 +1,286 @@
+            already_mutated_pos,
+            calc_future_dict=False,
+        )
+        second_mut_pos_feat_idx = self._get_mutating_pos_feat_idx(second_mutation_pos, flanked_seq)
+
+        return first_mut_pos_feat_idx, MultiFeatureMutationStep(
+            second_mut_pos_feat_idx,
+            second_mutation_pos,
+            neighbors_feat_old=feat_dict_future,
+            neighbors_feat_new=feat_dict_curr,
+        )
+
+    def update_mutation_step(
+            self,
+            mutation_step,
+            mutation_pos,
+            old_mutation_pos,
+            seq_mut_order,
+            intermediate_seq,
+            already_mutated_pos,
+            calc_future_dict=True,
+        ):
+        """
+        Does the heavy lifting for calculating feature vectors at a given mutation step
+        @param mutation_step: mutation step index
+        @param mutation_pos: the position that is mutating
+        @param old_mutation_pos: the position that mutated previously - None if this is first mutation
+        @param seq_mut_order: ImputedSequenceMutations
+        @param intermediate_seq: nucleotide sequence INCLUDING flanks - before the mutation step occurs
+
+        @return tuple with
+            1. the feature index of the position that mutated
+            2. a dict with the positions next to the previous mutation and their feature index
+            3. a dict with the positions next to the current mutation and their feature index
+        """
+        feat_dict_curr = dict()
+        feat_dict_future = dict()
+        # Calculate features for positions in the risk group at the time of this mutation step
+        # Only requires updating feature values that were close to the previous mutation
+        # Get the feature vectors for the positions that might be affected by the previous mutation
+        if old_mutation_pos is not None:
+            feat_dict_curr = self._get_feature_dict_for_region(
+                old_mutation_pos,
+                intermediate_seq,
+                seq_mut_order.obs_seq_mutation.seq_len,
+                already_mutated_pos,
+            )
+
+        # Calculate the features in these special positions for updating the next mutation step's risk group
+        # Get the feature vectors for the positions that will be affected by current mutation
+        if calc_future_dict:
+            feat_dict_future = self._get_feature_dict_for_region(
+                mutation_pos,
+                intermediate_seq,
+                seq_mut_order.obs_seq_mutation.seq_len,
+                already_mutated_pos,
+            )
+        return feat_dict_curr, feat_dict_future
+
+    def _get_mutated_seq(self, intermediate_seq, pos, end_seq):
+        """
+        @param intermediate_seq: initial nucleotide sequence
+        @param pos: mutating position
+        @param end_seq: final nucleotide sequence
+
+        @return a dict with the positions next to the given position and their feature index
```

Comment:
huh? you dont return a dict... im confused what this function is for. it takes in the initial string, the final string, and returns... basically the final string back?